### PR TITLE
Added h2 flat file importing and modified load_final_sumstats_mt to filter by h2 filters

### PR DESCRIPTION
**Pull request by rahulg603**
Added import_heritability which converts h2 flat files to a hail table, and includes loading functions for the h2 ht. Modified load_final_sumstats_mt to include a filter_pheno_h2_qc argument. Enabling filter_pheno_h2_qc filters pheno-ancestry pairs to include only those that pass the following filters: GWAS_run, defined_h2, significant_z, in_bounds_h2, normal_lambda, normal_ratio, EUR_plus_1.

### CODE REVIEW

File: heritability/import_heritability.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+def get_h2_ht():
+    return 'gs://ukb-diverse-pops/rg-h2-tables/ht/h2_estimates_all.ht'
+
+
+def import_h2_flat_file(save_to_ht, overwrite):
+    ht = hl.import_table(get_h2_flat_file(), 
+                         delimiter='\t', 
+                         impute=True, 
+                         key=['trait_type','phenocode','pheno_sex','coding','modifier'])
```
Comment (konradjk) [inline heritability/import_heritability.py]: neither here nor there but we do have this in a global variable somewhere in this repo, like `PHENO_KEY_FIELDS`

File: heritability/import_heritability.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    ht = ht.rename({'ancestry':'pop'})
+    ht = ht.drop('phenotype_id')
+    # solution from Zulip to munge flat file columns into nested structs
+    d = {}
+    row = ht.row
+    for name in row:
+        if name not in ht.key:
+            def recur(dict_ref, split_name):
```
Comment (konradjk) [inline heritability/import_heritability.py]: Can you pull these functions outside the for loop? Now that I'm saying it out loud, I don't know why we do that, but that is our typical convention 🤔

File: utils/results.py [hunk 3]
```diff
@@ -20,14 +21,29 @@ def filter_lambda_gc(lambda_gc):
...
+    def update_pheno_struct(pheno_struct, mt):
+        pheno_struct = pheno_struct.annotate(saige_heritability = pheno_struct.heritability,
+                                             heritability = mt.paired_pop_h2.get(pheno_struct.pop, 
+                                                                                 hl.missing(hl.tstruct(**mt.heritability[0].dtype))))
+        pheno_struct = pheno_struct.annotate(heritability = pheno_struct.heritability.drop('pop'))
+        return pheno_struct
+
+
+    mt = mt.annotate_cols(paired_pop_h2 = hl.dict(hl.map(lambda x: (x.pop,x), mt.heritability)))
+    mt = mt.annotate_cols(pheno_data = mt.pheno_data.map(lambda x: update_pheno_struct(x, mt)))
```
Comment (konradjk) [inline utils/results.py:45]: Combine these into 1!

File: utils/results.py [hunk 4]
```diff
@@ -42,6 +58,27 @@ def load_final_sumstats_mt(filter_phenos: bool = True, filter_variants: bool = T
         ).drop('pheno_indices')
         mt = mt.filter_cols(hl.len(mt.pheno_data) > 0)
 
+    if filter_pheno_h2_qc:
+        mt = mt.filter_cols(mt.pheno_data.any(lambda x: (~hl.is_missing(x.heritability.qcflags.pass_all)) & \
```
Comment (konradjk) [inline utils/results.py]: not `hl.is_defined`?

File: utils/results.py [hunk 5]
```diff
@@ -42,6 +58,27 @@ def load_final_sumstats_mt(filter_phenos: bool = True, filter_variants: bool = T
...
+                                              (x.heritability.qcflags.pass_all)))
+
+        mt = mt.annotate_cols(tf_filt_h2 = mt.pheno_data.map(lambda x: x.heritability.qcflags.pass_all))
+        # filtering column arrays
+        colfields_for_filt = ['pheno_data']
+        colexpr = {}
+        for x in colfields_for_filt:
+            this_col = hl.zip(mt[x], mt.tf_filt_h2).filter(lambda x: x[1]).map(lambda x: x[0])
+            colexpr.update({x: this_col})
```
Comment (konradjk) [inline utils/results.py]: Make this a function maybe, since you use it twice?

File: utils/results.py (hunk 3 above)
Comment (rahulg603) [reply utils/results.py:45]: update_pheno_struct depends on mt.paired_pop_h2 -- is that fine within a single annotate command?

Comment (konradjk) [reply utils/results.py:45]: Oh whoops, I didn't catch that - withdrawn!

File: utils/results.py (hunk 4 above)
Comment (rahulg603) [reply utils/results.py]: Changed ~hl.is_missing to hl.is_defined (these are identical I think right?)

File: utils/results.py (hunk 5 above)
Comment (rahulg603) [reply utils/results.py]: done

**Review by konradjk — APPROVED** [review]

**Comment by konradjk** [comment]
Oops, I approved, but probably remove the .DS_store file before you merge

**Pull request merged**